internal/provider/kubernetes/httproute.go

```diff
@@ -121,17 +121,61 @@ func (r *httpRouteReconciler) Reconcile(ctx context.Context, request reconcile.R
 
 	log.Info("reconciling httproute")
 
-	// Fetch the HTTPRoute from the cache.
-	httpRoute := &gwapiv1b1.HTTPRoute{}
-	err := r.client.Get(ctx, request.NamespacedName, httpRoute)
-	if errors.IsNotFound(err) {
-		log.V(2).Info("httproute not found, deleting it from the ResourceTable")
+	// Fetch all HTTPRoutes from the cache.
+	routeList := &gwapiv1b1.HTTPRouteList{}
+	if err := r.client.List(ctx, routeList); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error listing httproutes")
+	}
+
+	found := false
+	for i := range routeList.Items {
+		// Store the httproute in the resource map.
+		route := routeList.Items[i]
+		routeKey := NamespacedName(route.DeepCopy())
+		r.resources.HTTPRoutes.Store(routeKey, &route)
+		log.Info("added httproute to resource map")
+
+		if routeKey == request.NamespacedName {
+			found = true
+			// Get the route's namespace from the cache.
+			nsKey := types.NamespacedName{Name: route.Namespace}
+			ns := new(corev1.Namespace)
+			if err := r.client.Get(ctx, nsKey, ns); err != nil {
+				if errors.IsNotFound(err) {
+					// The route's namespace doesn't exist in the cache, so remove it from
+					// the namespace resource map if it exists.
+					if _, ok := r.resources.Namespaces.Load(nsKey.Name); ok {
+						r.resources.Namespaces.Delete(nsKey.Name)
+						log.Info("deleted namespace from resource map")
+					}
+				}
+				return reconcile.Result{}, fmt.Errorf("failed to get namespace %s", nsKey.Name)
+			}
+
+			// The route's namespace exists, so add it to the resource map.
+			r.resources.Namespaces.Store(routeKey.Namespace, ns)
+			log.Info("added namespace to resource map")
```

Comment:
Any reason not to run this logic for _all_ HTTPRoutes, not just the one that triggered the Reconcile? I think it'd be easier to reason about if we were processing all Resources, and likely helps avoid cold-start issues.